packages/langchain/src/openai/chat.ts

```diff
@@ -0,0 +1,74 @@
+import { CallbackManagerForLLMRun } from '@langchain/core/callbacks/manager';
+import { BaseMessage } from '@langchain/core/messages';
+import type { ChatResult } from '@langchain/core/outputs';
+import { AzureChatOpenAI, AzureOpenAI } from '@langchain/openai';
+import { OpenAiChatClient as OpenAiChatClientBase } from '@sap-ai-sdk/foundation-models';
+import { mapLangchainToAiClient, mapResponseToChatResult } from './util.js';
+import type { OpenAiChatModelInput, OpenAiChatCallOptions } from './types.js';
+
+/**
+ * OpenAI Language Model Wrapper to generate texts.
+ */
+export class OpenAiChatClient extends AzureChatOpenAI {
+  declare CallOptions: OpenAiChatCallOptions;
+  private openAiChatClient: OpenAiChatClientBase;
+
+  constructor(fields: OpenAiChatModelInput) {
+    const defaultValues = new AzureOpenAI({ apiKey: 'dummy' });
+    const stop = fields.stop
+      ? Array.isArray(fields.stop)
+        ? fields.stop
+        : [fields.stop]
+      : defaultValues.stop;
```

Comment:
Stop is handled differently because the types from langchain and us are crashing on a different level so to say.
All other types are matching the "real" type, but are actually optional.
Stop is not matching the "real" type (string | string[] | undefined) the superclass is inferring this type: (string[] | undefined), to match the superclass' type, we cast it to an array if it isn't one.

I agree I could use a util function for this though.